rita_common/src/debt_keeper/mod.rs

```diff
@@ -362,9 +371,9 @@ impl DebtKeeper {
 
     fn save(&mut self) -> Result<(), IOError> {
         // convert to the serializeable format and dump to the disk
-        let serialized = serde_json::to_string(&debt_data_to_ser(self.debt_data.clone()))?;
+        let serialized = bincode::serialize(&debt_data_to_ser(self.debt_data.clone())).unwrap();
         let mut file = File::create(settings::get_rita_common().payment.debts_file)?;
-        file.write_all(serialized.as_bytes())
```

Comment:
you seem to have modified only teh save routine, how will we load bincode encoded data if we're still trying to decode it using json? Also we have to take into account that what's on the disk is currently json and migrate it.

So there needs to be some logic that 

1) tries to deserialize with both json and bincode, takes what's successful (the json path goes away next release once every is migrated)
2) renames the file in the settings to .bincode rather than .json

Also save if needed should be called from the saving loop now so that we don't need to put the save frequency constants in this file and we can keep them in one place.